appimage/private/tool/mkappimage.py

```diff
@@ -187,6 +187,10 @@ def populate_appdir(appdir: Path, params: AppDirParams) -> None:
         # example entry: {"dst": "tests/test_py.runfiles/_main/tests/data.txt", "src": "tests/data.txt"}
         src = Path(file["src"]).resolve()
         dst = Path(appdir / file["dst"]).resolve()
+        if dst.exists():
+            # this is likely a runfile of a transitioned binary that's also present in untransitioned form.
+            # We shouldn't try to overwrite it because generated files are read-only.
+            continue
```

Comment:
How about something like
```suggestion
            if src.read_bytes() == dst.read_bytes():
                continue
            else:
                raise NotImplementedError(f"Got more than one {dst=} with different contents")
```